src/main/groovy/com/bmuschko/gradle/docker/tasks/image/Dockerfile.groovy

```diff
@@ -373,6 +373,30 @@ class Dockerfile extends DefaultTask {
         }
     }
 
+    static abstract class MapInstruction implements Instruction {
```

Comment:
Since the LABEL instruction accepts a series of key-value pairs, I decided to use a Map data type for representing this (I would have thought twice with plain Java, but the Groovy syntax for working with Maps is very clean and easy).

I wondered whether it makes sense to create a base class for this instruction type at all, since LABEL is (for now) the only Dockerfile instruction that works with key-value pairs.  However, I decided to aim for consistency with the other instruction types here... and who knows what additional instruction will be added to the Dockerfile spec in the future as it matures.